packages/agent/src/agent/http/index.ts

```diff
@@ -72,6 +72,13 @@ export interface HttpAgentOptions {
   // A surrogate to the global fetch function. Useful for testing.
   fetch?: typeof fetch;
 
+  // Additional options to pass along to fetch. Will not override fields that
+  // the agent already needs to set
+  fetchOptions?: Record<string, unknown>;
+
+  // Additional options to pass along to fetch for the call API.
+  callOptions?: Record<string, unknown>;
```

Comment:
I've attempted to create a minor unit test here. 
There is a repeated error that shows `(node:2473) V8: /home/runner/work/agent-js/agent-js/node_modules/borc/src/decoder.asm.js:3 Linking failure in asm.js: Unexpected stdlib member`.
Not sure but this might be due to issue here of jest globals and node globals differing. https://github.com/facebook/jest/issues/2549.

Reply:
Hmm, could be something like that. What Node.js version are you using?

The unit test works as expected on my laptop (equivalent to the [CI output](https://github.com/dfinity/agent-js/actions/runs/3484071076/jobs/5828221075)).

Reply:
I pushed some fixes for the unit test to hopefully save you a bit of time. Feel free to make any changes or improvements as you see fit.

Reply:
Thanks so much! I was using node version 16.17.0. (I'm still not sure how to get the unit test to work locally.) But I'm glad this fix works!